app-api/src/main/java/org/openmrs/eip/app/management/repository/DebeziumOffsetRepository.java

```diff
@@ -0,0 +1,23 @@
+package org.openmrs.eip.app.management.repository;
+
+import java.time.LocalDateTime;
+
+import org.openmrs.eip.app.management.entity.DebeziumOffset;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.transaction.annotation.Transactional;
+
+public interface DebeziumOffsetRepository extends JpaRepository<DebeziumOffset, Long> {
+	
+	long countByEnabledTrue();
+	
+	DebeziumOffset findFirstByEnabledTrueOrderByDateCreatedDesc();
+	
+	@Query(value = "UPDATE DebeziumOffset SET enabled = false, dateChanged = :dateChanged WHERE binlogFileName = :binlogFileName")
+	@Modifying
+	@Transactional
```

Comment:
Transactional methods make almost no sense on a JPA repository in this application